btree/btree.c

```diff
@@ -1926,9 +1961,15 @@ static int64_t tree_delete(struct node_op *op, struct td *tree,
 			   struct m0_be_tx *tx, int nxt)
 {
 	M0_PRE(tree != NULL);
-	return segops->so_tree_delete(op, tree, tx, nxt);
+	struct nd *root = tree->t_root;
+
+	op->no_tree = tree;
+	op->no_node = root;
+	node_free(op, op->no_node, tx, nxt);
```

Comment:
you have not called node_fini before freeing node. I think . it is required, please check it once.

Reply:
Yes. I may have missed that. Let me check.